### PR TITLE
add connection configuration when connect to zk

### PR DESCRIPTION
**Pull request by cyflhn**

### CODE REVIEW

File: examples/basic.go [hunk 1]
```diff
@@ -2,21 +2,53 @@ package main
...
-
-	"github.com/samuel/go-zookeeper/zk"
 )
 
 func main() {
-	c, _, err := zk.Connect([]string{"127.0.0.1"}, time.Second) //*10)
-	if err != nil {
-		panic(err)
+
+	servers := []string{"192.168.28.191:2181"}
```
Comment (ngaut) [inline examples/basic.go]: Why use 192.168.28.191

File: examples/basic.go [hunk 2]
```diff
@@ -2,21 +2,53 @@ package main
...
+
+	conf := zk.ConnConf{
+		RecvTimeout:    5 * time.Second,
+		ConnTimeout:    5 * time.Second,
+		SessionTimeout: 30000,
 	}
-	children, stat, ch, err := c.ChildrenW("/")
+
+	c, _, err := zk.ConnectWithConf(servers, conf)
 	if err != nil {
```
Comment (ngaut) [inline examples/basic.go:16]: Could you do some clean up.

File: zk/constants.go [hunk 3]
```diff
@@ -69,8 +69,9 @@ const (
 )
 
 const (
-	FlagEphemeral = 1
-	FlagSequence  = 2
+	FlagPersistent = 0
```
Comment (ngaut) [inline zk/constants.go:72]: Why  add FlagPersistent. I can't find any code reference it.

Comment (cyflhn) [inline zk/constants.go:72]: first thanks for replying me. I have been waiting for your reply for long time.  These flags can be used by other applications,such as codis. Let me show some codes in codis, "zkhelper.CreateRecursive(zkConn, dir, "", 0, zkhelper.DefaultDirACLs())"
"0" should be replaced by "FlagPersistent "

File: examples/basic.go (hunk 1 above)
Comment (cyflhn) [inline examples/basic.go]: The code is an example. So I select a zookeeper randomly for testing my code

**Comment by ngaut** [comment]
Could you clean up log and debug message ?

**Comment by cyflhn** [comment]
ok I have already removed debug logs

**Comment by cyflhn** [comment]
Is there any other modification should be done?

**Comment by ngaut** [comment]
Could you add some tests for ConnectWithConfDialer.

**Comment by cyflhn** [comment]
ok I have already added tests. I replaced the function "Connect" with "ConnectWithConfDialer" in server_helper.go

**Comment by ngaut** [comment]
Please reset to clean up commit messages. Then send another PR.
Thanks.

**Comment by cyflhn** [comment]
any other modification?

**Comment by cyflhn** [comment]
" reset to clean up commit messages" what does it mean?

**Comment by ngaut** [comment]
There are many ugly commit messages lie:
    add test    305d84c
 @cyflhn    Merge branch 'master' of https://github.com/cyflhn/go-zookeeper  a0e4437
 @cyflhn    add test case   4134962
 @cyflhn    Merge branch 'master' of https://github.com/cyflhn/go-zookeeper  d087a88
 @cyflhn    add test case   c7eeb32
 @cyflhn    Merge branch 'master' of https://github.com/cyflhn/go-zookeeper

Please do : 
git reset   init version of this fork
git commit 
git push
then send PR.

**Comment by cyflhn** [comment]
I have already sent a new PR

**Pull request closed without merging**